testing/sdk_auth.py

```diff
@@ -282,13 +287,37 @@ def create_remote_keytab(self, name: str, principals: list=[]) -> str:
                                             "runs/latest", name)
         return keytab_absolute_path
 
+
+    @retrying.retry(stop_max_attempt_number=6, wait_fixed=60000)
     def get_keytab_for_principals(self, principals: list, output_filename: str):
         """
         Download a generated keytab for the specified list of principals
         """
         remote_keytab_path = self.create_remote_keytab(self.keytab_file_name, principals=principals)
         _copy_file_to_localhost(self.kdc_host_id, remote_keytab_path, output_filename)
 
+        # In a fun twist, at least in the HDFS tests, we sometimes wind up with a _bad_ keytab. I know, right?
+        # We can validate if it is good or bad by checking it for a certain pattern. That pattern, is when
+        # strings is run against it (remember, strings will show any 4+ consecutive ASCII characters by default)
+        # that some lines will start with ZX.
+        #
+        # See HDFS-493 if you'd like to learn more. Personally, I'd like to forget about this.
+        command = "strings {} | grep ^ZX".format(output_filename)
+        result = subprocess.run(command,
+               shell=True,
+               stdout=subprocess.PIPE,
+               stderr=subprocess.PIPE)
+
+        # Grep exits 0 if there were matches
+        if result.returncode == 0:
+            log.info("There were some matches when checking the keytab for ^ZX: %s", result.stdout)
+            # Clean up the bad keytab.
+            sdk_tasks.task_exec(self.kdc_host_id)
```

Comment:
This command doesn't seem to do anything? (I see the deletion happens in the creation step anyway). Also, wasn't `task_exec` moved to `sdk_cmd`?

Reply:
Whoops. I started to type a thing, but then didn't Thanks.